products.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs').promises;
+const path = require('path');
+const productsFile = path.join(__dirname, 'data/full-products.json');
+
+// List all products with pagination and filtering support
+async function list(options = {}) {
+  const { offset = 0, limit = 25, tag } = options;
+  const data = await fs.readFile(productsFile);
+  const products = JSON.parse(data);
+
+  let filteredProducts = products;
+  if (tag) {
+    filteredProducts = products.filter(product => product.tags.includes(tag));
```

Comment:
This won't work, only because the `product.tags` array is an array of objects, not strings while tag is a string.

So given the array below, the tags array will never return true for `tags.include("tag1")`
```js
let tags = [
{ title: "tag1", id:1 },
{ title: "tag2", id:2 }
];
```